batch/batch/aioclient.py

```diff
@@ -12,6 +12,23 @@
 
 
 class Job:
+    @staticmethod
+    def exit_code(job_status):
+        if 'exit_code' in job_status and job_status['exit_code'] is not None:
```

Comment:
Minor thing: I don't prefer the this style where the entire function body is in an if.  Better to flip the sense of the if and get out early.

LLVM has the same coding guideline which I like: https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code